sphinx/builders/html/__init__.py

```diff
@@ -982,9 +982,9 @@ def post_process_images(self, doctree: Node) -> None:
                 node.replace_self(reference)
                 reference.append(node)
 
-    def load_indexer(self, docnames: Iterable[str]) -> None:
+    def load_indexer(self, docnames: Set[str]) -> None:
         assert self.indexer is not None
-        keep = set(self.env.all_docs) - set(docnames)
+        keep = set(self.env.all_docs) - docnames
```

Comment:
I would still keep an iterable, just in case, or maybe keep the `set()`. Or just write `set(...).difference(docnames)`. That way, you have the correct typing *and* you won't have possible runtime issues.